### ISSUE TITLE
Bypassing face segmentation module?

### ISSUE THREAD

**Issue by sbanerj1** [YuvalNirkin/face_swap#30]
Hi Yuval,

Thanks for sharing the code for your paper, really appreciate it. I am planning to use the results for a paper I am writing. However, I am unable to compile all the necessary dependencies correctly, mostly due to the face segmentation module. Is it possible to bypass the face segmentation module and provide the network pre-computed segmentation masks instead? This will be easier in my case as there is a python version of your segmentation model available (here: https://github.com/shaoanlu/face-segmentation-keras). 

Also, is there a possibility of a python release of your model in the near future?

Regards,
Sandipan

**Comment by YuvalNirkin** [comment]
Hi Sandipan,
The code does support loading external segmentations but currently you have to build the face segmentation module as well. 
For which dataset do you need pre-computed segmentation masks?

I am working on adding Python binding to make the code more accessible, and also a PyTorch version of the face segmentation module.

Best,
Yuval

**Comment by sbanerj1** [comment]
Hi Yuval,

Thanks for your fast reply. I have segmentation masks on LFW and IJB-B but the problem is the face segmentation module and its dependencies for me while building.  
Is there a possibility that the pytorch version will be released in the next couple of weeks or so? Otherwise, we might have to discuss how to generate the results for my paper.

Regards,
Sandipan

**Comment by YuvalNirkin** [comment]
I am not sure I'll be able to release the pytorch version so soon.
But I have updated the code, I removed some of the dependencies to make it easier to build and also added Python binding. Try to follow this detailed [Ubuntu installation guide](https://github.com/YuvalNirkin/face_swap/wiki/Ubuntu-Installation-Guide).
